File: Fantasy.Base/ServiceModel/WCFExceptionHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ServiceModel;

namespace Fantasy.ServiceModel
{
    public static class WCFExceptionHandler
    {

        private static Type[] _knowns = new Type[] 
        {
            typeof(EndpointNotFoundException),
            typeof(TimeoutException),
            typeof(CommunicationException),
            typeof(ServerTooBusyException),
            typeof(CommunicationObjectAbortedException),
            typeof(CommunicationObjectFaultedException),
            typeof(ObjectDisposedException), 
            typeof(CallbackExpiredException) 
        };

        public static void CatchKnowns(Exception error)
        {
            Type et = error.GetType();

            if (et.IsGenericType && et.GetGenericTypeDefinition() == typeof(FaultException<>))
            {
                et = et.GetGenericArguments()[0];
            }


            if (Array.IndexOf(_knowns, et) < 0)
            {
                throw error;
            }
        }

        public static bool CanCatch(Exception error)
        {
            Type et = error.GetType();

            if (et.IsGenericType && et.GetGenericTypeDefinition() == typeof(FaultException<>))
            {
                et = et.GetGenericArguments()[0];
            }

            return Array.IndexOf(_knowns, et) >= 0;
            
        }
    }
}

File: branches/Stheno/AutoConverted/Fantasy.Base/ServiceModel/WCFExceptionHandler.java
package Fantasy.ServiceModel;

public final class WCFExceptionHandler
{

	private static java.lang.Class[] _knowns = new java.lang.Class[] { EndpointNotFoundException.class, TimeoutException.class, CommunicationException.class, ServerTooBusyException.class, CommunicationObjectAbortedException.class, CommunicationObjectFaultedException.class, ObjectDisposedException.class, CallbackExpiredException.class };

	public static void CatchKnowns(RuntimeException error)
	{
		java.lang.Class et = error.getClass();

		if (et.IsGenericType && et.GetGenericTypeDefinition() == FaultException<>.class)
		{
			et = et.GetGenericArguments()[0];
		}


		if (Array.indexOf(_knowns, et) < 0)
		{
			throw error;
		}
	}

	public static boolean CanCatch(RuntimeException error)
	{
		java.lang.Class et = error.getClass();

		if (et.IsGenericType && et.GetGenericTypeDefinition() == FaultException<>.class)
		{
			et = et.GetGenericArguments()[0];
		}

		return Array.indexOf(_knowns, et) >= 0;

	}
}